src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -25,6 +27,88 @@ export class AccountsBalanceInfoService extends AbstractService {
 	): Promise<IAccountBalanceInfo> {
 		const { api } = this;
 
+		// Retrieve the historic api for the given hash
+		const historicalApi = await api.at(hash);
+
+		/**
+		 * Check two different cases where a historicalApi is needed in order
+		 * to have the correct runtime methods.
+		 *
+		 * a) Does the block use the oldest api where the free balance is found
+		 * using `historicalApi.query.balances.freeBalance`
+		 *
+		 * b) Does the block use an older api where the free balance is within the
+		 * AccountInfo type, but the storage does not yet have the `.at` method.
+		 */
+		if (historicalApi.query.balances.freeBalance) {
+			const [header, free, locks, reserved, nonce] = await Promise.all([
+				api.rpc.chain.getHeader(hash),
+				historicalApi.query.balances.freeBalance(address) as Promise<Balance>,
+				historicalApi.query.balances.locks(address),
+				historicalApi.query.balances.reservedBalance(
+					address
+				) as Promise<Balance>,
+				historicalApi.query.system.accountNonce(address) as Promise<Index>,
+			]);
+
+			// Values dont exist for these historic runtimes
+			const miscFrozen = api.registry.createType('Balance', 0),
+				feeFrozen = api.registry.createType('Balance', 0);
+
+			const at = {
+				hash,
+				height: header.number.toNumber().toString(10),
+			};
+
+			if (locks && free && reserved && nonce) {
```

Comment:
I wonder if this is necessary, could we just do null if one of these fields doesn't exist?

Reply:
When you say 'do null' do you mean a `null` response for the key value pair? Or do you mean just check one value and then throw the error. 

Here I am just mimicking the error response logic that is in the existing service to keep it consistent with what response/errors it reports. So if we change this to something different it might be worth updating the rest as well.

Reply:
Ahh, I see what you are saying, It only makes sense to really check for one value. Will update